src/bentoml/_internal/io_descriptors/json.py

```diff
@@ -192,6 +265,15 @@ def __init__(
                 pydantic_model, pydantic.BaseModel
             ), "'pydantic_model' must be a subclass of 'pydantic.BaseModel'."
 
+        if typeddict is not None and not is_typeddict(typeddict):
+            raise BadInput("'typeddict' must be inherited 'TypedDict'.")
+
+        if is_legacy_typeddict(typeddict):
+            raise BadInput(
+                "You should use `typing_extensions.TypedDict` instead of `typing.TypedDict` with Python < 3.11."
+            )
```

Comment:
A quick refactor, imho:
```suggestion
        if typeddict is not None and not is_typeddict(typeddict):
            	raise BadInput(f"'typeddict={typeddict}' is not a subclass of '{TypedDict}'.")
```

I don't think asking user to use `typing_extensions` to create TypedDict is sensible here.

Since we are going to support 3.8, 3.9 and 3.10 for a while (at least until it reaches EOL), we then should also support "legacy" TypedDict

Also What is the difference between legacy and the "new" TypedDict may I ask?

Reply:
[feat: remove legacy typeddict check (56e3763)](https://github.com/bentoml/BentoML/pull/3809/commits/56e3763f193481793b4a40b2a02293a8a8fcf04c)